arctic/_cache.py

```diff
@@ -3,43 +3,84 @@
 
 from pymongo.errors import OperationFailure
 
-from ._config import CACHE_COLL, CACHE_DB
-
 logger = logging.getLogger(__name__)
 
+CACHE_COLL = 'cache'
+CACHE_DB = 'meta_db'
+CACHE_SETTINGS = 'cache_settings'
+"""
+Sample cache_settings collection entry:
+meta_db.cache_settings.insertOne({"enabled": true, "cache_expiry": 600})
+meta_db.cache_settings.find(): { "_id" : ObjectId("5cd5388b9fddfbe6e968f11b"), "enabled" : false, "cache_expiry" : 600 }
+"""
+DEFAULT_CACHE_EXPIRY = 3600
+
 
 class Cache:
-    def __init__(self, client, cache_expiry=3600, cache_db=CACHE_DB, cache_col=CACHE_COLL):
+    def __init__(self, client, cache_expiry=DEFAULT_CACHE_EXPIRY, cache_db=CACHE_DB, cache_col=CACHE_COLL):
         self._client = client
         self._cachedb = client[cache_db]
         self._cachecol = None
         try:
             if cache_col not in self._cachedb.list_collection_names():
                 self._cachedb.create_collection(cache_col).create_index("date", expireAfterSeconds=cache_expiry)
         except OperationFailure as op:
-            logging.debug("This is fine if you are not admin. The collection should already be created for you: %s", op)
+            logging.info("This is fine if you are not admin. The collection should already be created for you: %s", op)
 
         self._cachecol = self._cachedb[cache_col]
 
-    def get(self, key, newer_than_secs=-1):
+    def _get_cache_settings(self):
+        try:
+            return self._cachedb[CACHE_SETTINGS].find_one()
+        except OperationFailure as op:
+            logging.debug("Cannot access %s in db: %s. Error: %s" % (CACHE_SETTINGS, CACHE_DB, op))
+        return None
+
+    def set_caching_state(self, enabled):
+        """
+        Used to enable or disable the caching globally
+        :return:
+        """
+        if not isinstance(enabled, bool):
+            logging.error("Enabled should be a boolean type.")
+            return
+
+        if CACHE_SETTINGS not in self._cachedb.list_collection_names():
+            logging.info("Creating %s collection for cache settings" % CACHE_SETTINGS)
+            self._cachedb[CACHE_SETTINGS].insert_one({
+                'enabled': enabled,
+                'cache_expiry': DEFAULT_CACHE_EXPIRY
+            })
+        else:
+            self._cachedb[CACHE_SETTINGS].update_one({}, {'$set': {'enabled': enabled}})
```

Comment:
This forces `cache_settings` to only ever have a single document.

My vision was to have a `meta_db.global_settings` collection. In here you can store individual documents for the settings you want:
```
{
  _id: 'library_cache_enabled',
  value: False
},
{
  _id: 'library_cache_expiry',
  value: 600
}
```

Then we can put all global settings into a single collection. What do you think?

Reply:
So I generally like having one document for a setting as this is read fairly often so I didn't want to get into writing / reading atomically and having 2 reads instead of 1. But yeah, I could have one settings doc per application in the same global_settings collection. Your call. Don't feel too strongly about it. 

I can change the name to just meta_db.settings for now and move throttling to use this collection later?